### PR TITLE
Support for ratings & reviews in Storefront App

### LINKED ISSUE #351: Support for ratings & reviews in Storefront App
As a Magento Developer, 

I would like GrraphQL to return product ratings and reviews data in a highly performant and scalable way by building the feature in the Catalog Storefront app

So that

Merchants who install Catalog Storefront app can continue to show the product ratings and reviews in a performant way.

<ins>**Acceptance Criteria**</ins>
* Catalog Storefront only support READ operations
* Review creation will continue to use the existing GraphQL endpoint
* Storefront app has the following data for ratings & reviews
  * Review Count
  * Reviews
    * Pagination support

Tech details:
* The following GQL Sceham should be exposed to Export API/SF API app/code/Magento/ReviewGraphQl/etc/schema.graphqls
  * Only Review/Rating for READ operations should be handled by Export API/SF API
  * Do not expose Review Count in Export API
  * Review Count must be calculated (aggregation Query to ES index) during request to SF API
* Export API should have a separate feed for Product Reviews
* Pagination should be available on SF API side

For Grooming:
as an output we should receive read-to-use :
* et_schema (for Export API)
* proto schema with services to obtain review/rating data

### PR DESCRIPTION
**Pull request by aleksejsbaranovs10**
### Related Pull Requests
https://github.com/magento/storefront-product-reviews-ce/pull/1
https://github.com/magento/catalog-storefront/pull/415

### Fixed Issues (if relevant)

1. Fixes magento/catalog-storefront#351

### Code Review Checklist (*)

- [ ] Story AC is completed
- [ ] proposed changes correspond to [Magento Technical Vision](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)
- [ ] new or changed code is covered with web-api/integration tests (if applicable)
- [ ] no backward incompatible changes

### CODE REVIEW

**Comment by aleksejsbaranovs10** [comment]
@magento run all tests

**Comment by aleksejsbaranovs10** [comment]
@magento run all tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Static Tests, WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Static Tests, WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Static Tests, WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Static Tests, WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Static Tests, WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run all tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run all tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Static Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by aleksejsbaranovs10** [comment]
@magento run WebAPI Tests

**Comment by RuslanKostiv1** [comment]
@aleksejsbaranovs10 please fix merge conflict

**Comment by RuslanKostiv1** [comment]
@magento run WebAPI Tests

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@magento run WebAPI Tests

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
Fix static failures
https://github.com/magento/commerce-data-export/pull/2/checks?check_run_id=1426480652 @aleksejsbaranovs10 
WebAPI Tests are fixed now

File: app/code/Magento/CatalogExport/Model/RatingMetadataRepository.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        $this->feedPool = $feedPool;
+        $this->ratingMetadataFactory = $ratingMetadataFactory;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->exportConfiguration = $exportConfiguration;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get(EntityRequest $request)
```
Comment (RuslanKostiv1) [inline app/code/Magento/CatalogExport/Model/RatingMetadataRepository.php]: return type is missing

File: app/code/Magento/CatalogExportApi/Api/RatingMetadataRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    /**
+     * Get rating metadata by rating ids
+     *
+     * @param \Magento\CatalogExportApi\Api\EntityRequest $request
+     *
+     * @return \Magento\CatalogExportApi\Api\Data\RatingMetadata[]
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function get(EntityRequest $request);
```
Comment (RuslanKostiv1) [inline app/code/Magento/CatalogExportApi/Api/RatingMetadataRepositoryInterface.php]: return type is missing

File: app/code/Magento/CatalogExportApi/Api/ReviewRepositoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    /**
+     * Get reviews by ids
+     *
+     * @param \Magento\CatalogExportApi\Api\EntityRequest $request
+     *
+     * @return \Magento\CatalogExportApi\Api\Data\Review[]
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function get(EntityRequest $request);
```
Comment (RuslanKostiv1) [inline app/code/Magento/CatalogExportApi/Api/ReviewRepositoryInterface.php]: return type is missing

**Comment by RuslanKostiv1** [comment]
Please fix Static Tests failures 
https://github.com/magento/commerce-data-export/pull/2/checks?check_run_id=1426480652

File: app/code/Magento/CatalogExport/Model/RatingMetadataRepository.php (hunk 1 above)
Comment (aleksejsbaranovs10) [reply app/code/Magento/CatalogExport/Model/RatingMetadataRepository.php]: fixed

File: app/code/Magento/CatalogExportApi/Api/RatingMetadataRepositoryInterface.php (hunk 2 above)
Comment (aleksejsbaranovs10) [reply app/code/Magento/CatalogExportApi/Api/RatingMetadataRepositoryInterface.php]: fixed

File: app/code/Magento/CatalogExportApi/Api/ReviewRepositoryInterface.php (hunk 3 above)
Comment (aleksejsbaranovs10) [reply app/code/Magento/CatalogExportApi/Api/ReviewRepositoryInterface.php]: fixed

**Comment by RuslanKostiv1** [comment]
@magento import pr to magento-commerce/commerce-data-export

**Comment by magento-engcom-team** [comment]
@RuslanKostiv1 the pull request successfully imported.

**Pull request merged**